packages/cubejs-materialize-driver/src/MaterializeDriver.ts

```diff
@@ -0,0 +1,102 @@
+import { PostgresDriver, PostgresDriverConfiguration } from '@cubejs-backend/postgres-driver';
+import { DownloadTableMemoryData, IndexesSQL, StreamOptions, StreamTableDataWithTypes, TableStructure } from '@cubejs-backend/query-orchestrator';
+import { PoolClient } from 'pg';
+import Stream from './Stream';
+
+export class MaterializeDriver extends PostgresDriver {
+  public constructor(options: PostgresDriverConfiguration) {
+    super(options);
+  }
+
+  protected async prepareConnection(
+    conn: PoolClient
+  ) {
+    await conn.query(`SET TIME ZONE '${this.config.storeTimezone || 'UTC'}'`);
+    // Support for statement_timeout is still pending. https://github.com/MaterializeInc/materialize/issues/10390
+  }
+
+  protected async loadUserDefinedTypes(): Promise<void> {
+    // Support for typcategory field still pending: https://github.com/MaterializeInc/materialize/issues/2157
+  }
+
+  /**
+   * @param {string} schemaName
+   * @return {Promise<Array<unknown>>}
+   */
+  public async createSchemaIfNotExists(schemaName: string): Promise<unknown[]> {
+    return this.query(
+      `SHOW SCHEMAS WHERE name = '${schemaName}'`, []
+    ).then((schemas) => {
+      if (schemas.length === 0) {
+        return this.query(`CREATE SCHEMA IF NOT EXISTS ${schemaName}`, []);
+      } else return [];
+    });
+  }
+
+  public async uploadTableWithIndexes(
```

Comment:
This looks like a clone of `BaseDriver.uploadTableWithIndexes`. Delete?

Reply:
I've replaced the code with `BaseDriver.prototype.uploadTableWithIndexes.apply(this, [table, columns, tableData, indexesSql, [], null]);` because parent class `PostgresDriver` uses `UNNEST` an is unsupported in Materialize.

I'm not 100% sure about the approach of using the prototype, but I'm highly open to any suggestions.

Reply:
I see. How about`BaseDriver.prototype.uploadTableWithIndexes.bind(this)(table, columns, tableData, indexesSql, [], null)`?